src/cascadia/Remoting/Monarch.cpp

```diff
@@ -102,17 +100,71 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
     // - the peasant if it exists in our map, otherwise null
     Remoting::IPeasant Monarch::_getPeasant(uint64_t peasantID)

```

Comment:
Maybe `std::optional<Remoting::IPeasant>` ?

Reply:
Nah. When you have a type that is already nullable, it only confuses things to make it optional as well. Optional adds nullability to non-nullable types. 

You end up in this situation:

```cpp
optional<IUnknown> t;
t = nullopt; // empty
t = null; // NOT EMPTY
t = object; // NOT EMPTY
```

Reply:
That's fair. The name `IPeasant` somehow alsoe indicates its nullability.